src/bootstrap/compile.rs

```diff
@@ -658,6 +658,12 @@ impl Step for Rustc {
 
         // With LLD, we can use ICF (identical code folding) to reduce the executable size
         // of librustc_driver/rustc and to improve i-cache utilization.
+        //
+        // -Wl,[link options] doesn't work on Windows. However, /OPT:ICF (technically /OPT:REF,ICF)
+        // is already on by default in Windows optimized builds, which is interpreted as --icf=all:
+        // https://github.com/llvm/llvm-project/blob/3329cec2f79185bafd678f310fafadba2a8c76d2/lld/COFF/Driver.cpp#L1746
+        // https://github.com/rust-lang/rust/blob/f22819bcce4abaff7d1246a56eec493418f9f4ee/compiler/rustc_codegen_ssa/src/back/linker.rs#L827
+        #[cfg(not(windows))]
```

Comment:
This is the platform of the *host* we are building on. But it's possible to cross compile from Unix to Windows, in which case I think we still use the MSVC toolchain.

Also, I think this needs to only special case MSVC - I think pc-windows-gnu should support these flags (@workingjubilee @Kobzol can you confirm?).

As a general rule, the only platform-specific `cfg` code in bootstrap should be related to filesystem stuff, everything else should be based off `builder.target`.

Reply:
I am very unsure of myself, but I _think_ that I should be checking `compiler.host.contains("msvc")` rather than `target.contains("msvc")` - the comment at the top of this function says "This will build the compiler for a particular stage of the build using the `compiler` targeting the `target` architecture", so I think we should be using the flag format for `compiler`. Let me know if I'm wrong!

Reply:
so, you're right that the distinction matters, but unfortunately bootstrap is really bad at naming things :sweat_smile: and `compiler` is "the compiler building `Rustc`", not "the compiler generated by the `Rustc` step". so `target` is correct.

If it helps, I double-checked this for myself by looking at what platform we pass to cargo, and it is `target`, not `compiler.host`: https://github.com/rust-lang/rust/blob/dcead65e48bb4957f9f20bdb037806dfa6c2db66/src/bootstrap/compile.rs#L650

Reply:
ok it turns out you are correct after all, we use the linker of the host and not the target. sorry for the confusion.